development/image-url-helper/pkg/missing/missing.go

```diff
@@ -0,0 +1,55 @@
+package missing
+
+import (
+	"strings"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/kyma-project/test-infra/development/image-url-helper/pkg/common"
+)
+
+// CheckForMissingImages checks if images exist and returns ComponentImageMap of nonexistent images
+func CheckForMissingImages(allImages common.ComponentImageMap, missingImages common.ComponentImageMap) error {
+
+	for imageURL, image := range allImages {
+		missing, err := isImageMissing(image)
+		if err != nil {
+			return err
+		}
+
+		if missing {
+			if !strings.Contains(err.Error(), "Failed to fetch") {
+				// unknown error, fail here
+				return err
+			}
+
+			// failed to fetch, add to list of non-existent images
+			componentNames := make([]string, 0)
+			for component := range image.Components {
+				componentNames = append(componentNames, component)
```

Comment:
In line 27 you create new array for each image. Why you rewrite image.Components array to the componentsNames array? Both will have the same values. You even doesn't pass this componentNames array outside this function.